### PR TITLE
Added docker pushing logs to cloudwatch functionality

### PR DESCRIPTION
**Pull request by bshien**
### Description

1. Changes docker log driver to awslogs so docker can push logs to cloudwatch.
2. Tuned canary alarm settings
3. Added and refactored tests for NginxProxyReadonly
4. Refactored SecretStack name

Logs look like this in Cloudwatch:
Screenshot 2024-06-26 at 3 51 48 PM

### Issues Resolved

https://github.com/opensearch-project/opensearch-metrics/issues/47

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Thanks @bshien, if possible can we remove the special characters in the log next to INFO and DEBUG?

File: infrastructure/lib/stacks/secrets.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export interface SecretProps {
     readonly secretName: string
 }
 
-export class OpenSearchMetricsSecrets extends Stack {
+export class OpenSearchMetricsSecretsStack extends Stack {
```
[the remark this answers is not in the document]
Comment (bshien) [reply infrastructure/lib/stacks/secrets.ts:9]: Since the other stacks have the Stack suffix, I thought it would be better to keep that style

**Comment by prudhvigodithi** [comment]
> Thanks @bshien, if possible can we remove the special characters in the log next to INFO and DEBUG?

Having a log formatter to `awssigv4proxy` should help fix the special characters.
https://forums.docker.com/t/how-to-turn-off-the-log-color-mark-output-by-stdout/117429

File: infrastructure/lib/stacks/secrets.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply infrastructure/lib/stacks/secrets.ts:9]: does this name change destroy the stack and recreate ?

[the remark this answers is not in the document]
Comment (bshien) [reply infrastructure/lib/stacks/secrets.ts:9]: Tested on local, it didn't destroy the stack, I think that only happens if the id changes

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Merging this as there is some issue with mend and not related to this code change.
Thanks

**Pull request merged**